consumer_group.go

```diff
@@ -864,6 +864,23 @@ func newConsumerGroupSession(ctx context.Context, parent *consumerGroup, claims
 	// start consuming
 	for topic, partitions := range claims {
 		for _, partition := range partitions {
+			if parent.client.ParttionNotReadable(topic, partition) {
+				// partition not readable, wait for it to become readable
+				go func(topic string, partition int32) {
+					for {
+						if parent.client.ParttionNotReadable(topic, partition) {
+							time.Sleep(5 * time.Second)
+						} else {
+							break
+						}
+					}
```

Comment:
[simplify] We can loop directly on conditional:
```suggestion
					for parent.client.ParttionNotReadable(topic, partition) {
						time.Sleep(5 * time.Second)
					}
```

Additionally, there are concerns about spinning up a goroutine without a clear lifetime. As well, that this seems to be an indefinite retry, potentially leaving the consumer in a degraded state without any clarity that it is retrying.